test/integration_test_helper.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module KubernetesDeploy
+  class IntegrationTest < KubernetesDeploy::TestCase
+    include KubeclientHelper
+    include FixtureDeployHelper
+
+    TestProvisioner.prepare_cluster
```

Comment:
This is the most important part. With this call moved into a file we only load when running the integration suite, we no longer make unnecessary API calls during unit tests. I'm still doing it on file load because it should happen only once even if we're running the parallel integration suite. Can you think of a nicer way?

Reply:
I'd probably have put this into some kind of `IntegrationTest#before_setup` method (and then memoize that it run to make sure it doesn't run more than once). Maybe unlikely but in theory you can load the code for the integration tests but still only run the unit tests, and in that case your strategy here wouldn't work (because it assumes that integration tests run if and only if they were loaded).

Reply:
Isn't that memoization ineffective if we're running tests in parallel, given that the `prepare_cluster` operation can take some time? I guess I can use a mutex to address that, but seems a bit complicated compared to the assumption that this file won't be loaded by unit tests (which as you say is probably pretty safe in practice).

I checked whether minitest has a `before_all`, and apparently the maintainers refused to provide such a hook because you can do what we're doing here instead: https://github.com/seattlerb/minitest/issues/61